### PR TITLE
 FEATURE: Set crop image adjustment by aspect ratio

### PR DESCRIPTION
**Pull request by robertlemke**
This change introduces a new property "aspectRatio" for the crop image
adjustment. It allows users to set a cropping area simply by providing
an aspect ratio, instead of x, y, width and height.

If an aspect ratio is specified, the x, y, width, and height parameters
are automatically deactivated. Likewise, if x, y, width or height are
specified, a potentially defined aspect ratio value will be reset.

### CODE REVIEW

File: Neos.Media/Migrations/Mysql/Version20190315122900.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?php
+namespace Neos\Flow\Persistence\Doctrine\Migrations;
```
Comment (daniellienert) [inline Neos.Media/Migrations/Mysql/Version20190315122900.php]: The pgsql migrations seems missing

File: Neos.Media.Browser/Classes/Controller/AssetController.php [hunk 2]
```diff
@@ -634,35 +661,39 @@ public function relatedNodesAction(AssetInterface $asset)
      * @return void
      * @Flow\Validate(argumentName="label", type="NotEmpty")
      * @Flow\Validate(argumentName="label", type="Label")
+     * @throws ForwardException
      */
-    public function createTagAction($label)
+    public function createTagAction($label): void
```
Comment (daniellienert) [inline Neos.Media.Browser/Classes/Controller/AssetController.php]:
```suggestion
    public function createTagAction(string $label): void
```

File: Neos.Media/Migrations/Mysql/Version20190315122900.php (hunk 1 above)
Comment (kdambekalns) [reply Neos.Media/Migrations/Mysql/Version20190315122900.php]: No longer true… :)

File: Neos.Media/Classes/Domain/Model/Adjustment/CropImageAdjustment.php [hunk 3]
```diff
@@ -140,29 +151,90 @@ public function setY($y)
...
+     * @return bool
      */
-    public function canBeApplied(ImagineImageInterface $image)
+    public function canBeApplied(ImagineImageInterface $image): bool
     {
-        if (
+        if ($this->aspectRatioAsString !== null) {
+            $desiredAspectRatio = AspectRatio::fromString($this->aspectRatioAsString);
+            $originalAspectRatio = new AspectRatio($image->getSize()->getWidth(), $image->getSize()->getHeight());
+            return $originalAspectRatio->getRatio() !== $desiredAspectRatio->getRatio();
```
Comment (kitsunet) [inline Neos.Media/Classes/Domain/Model/Adjustment/CropImageAdjustment.php]: wait, why does the aspect ratio need to be different. What if I wanted to crop a smaller part but with the same aspect ratio?

Comment (kitsunet) [reply Neos.Media/Classes/Domain/Model/Adjustment/CropImageAdjustment.php]: Ok, I wrote this comment before writing the overall conclusion above. It's pretty much clear now, jsut reinforces that it doesn't make much sense to me.

**Pull request merged**